### PR TITLE
Call for notification with payment

### PR DESCRIPTION
**Pull request by Gabriiii99**
## Short description

Implemented a call to create a notification with payment

## List of changes proposed in this pull request

- Improved the model class to create a JSON for the notifications with payment
- Implemented test 10394 that use the new call

## How to test

Simply run `mvn clean test -D"cucumber.filter.tags=@annullamentoNotifica"` and verify if the test works fine.

- [ ] Does this PR require a manual test in AWS CodeBuild?

### CODE REVIEW

File: src/main/java/it/pn/frontend/e2e/model/NotificationMetadataAttachment.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    public NotificationMetadataAttachment() {
+        this.digests = new NotificationAttachmentDigests("UtTwRseo3KHQIFkl7+VGNeZDWauWqYyDEjrxl9w/h2g=");
+        this.contentType = "application/json";
+        this.ref = new NotificationAttachmentBodyRef("PN_F24_META-5cf8e14c93c34ed194ff3ace14620fc1.json", "v1");
+    }
+
+    public NotificationMetadataAttachment(String sha256, String key, String versionToken) {
+        this.digests = new NotificationAttachmentDigests(sha256);
+        this.contentType = "application/json";
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/model/NotificationMetadataAttachment.java]: take contentType final

File: src/main/java/it/pn/frontend/e2e/model/NotificationPaymentAttachment.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    public NotificationPaymentAttachment() {
+        this.digests = new NotificationAttachmentDigests("1QKD/Ks6BohyQ+bgMxHf9NrpNhVmGUPxRYE1aerU4JQ=");
+        this.contentType = "application/pdf";
+        this.ref = new NotificationAttachmentBodyRef("PN_NOTIFICATION_ATTACHMENTS-1eaca1b63f8a4c519d1705c29e83b59c.pdf", "v1");
+    }
+
+    public NotificationPaymentAttachment(String sha256, String key, String versionToken) {
+        this.digests = new NotificationAttachmentDigests(sha256);
+        this.contentType = "application/pdf";
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/model/NotificationPaymentAttachment.java]: as before

File: src/test/resources/feature/2-mittente/3_downloadFile/014_1_downloadFileDettaglioNotificaMittente.feature [hunk 3]
```diff
@@ -8,6 +8,9 @@ Feature: Mittente scarica tutti i file all'interno di una notifica
   Scenario: PN-9327 - Mittente scarica attestazioni
     Given PA - Si effettua la login tramite token exchange, e viene visualizzata la dashboard
     When Creo in background una notifica con un destinatario e un documento tramite API REST
+      | avvisoPagoPa | false |
+      | F24          | false |
+      | entrambi     | true  |
     Then Attendo 5 minuti e verifico in background che la notifica sia stata creata correttamente
```
Comment (leleOFA) [inline src/test/resources/feature/2-mittente/3_downloadFile/014_1_downloadFileDettaglioNotificaMittente.feature:14]: insert here polling for verify create notification

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java [hunk 4]
```diff
@@ -157,8 +157,8 @@ public boolean titoloDiPagamentoDisplayed() {
     public boolean codiceAvvisoDisplayed() {
         try {
             getWebDriverWait(5).withMessage("Il sezione codice avviso non è visibile").until(ExpectedConditions.visibilityOf(codiceAvviso)).isDisplayed();
-        return true;
-        }catch (RuntimeException e){
+            return true;
+        } catch (RuntimeException e) {
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java]: delete try catch

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java [hunk 5]
```diff
@@ -171,14 +171,14 @@ public boolean scaricaAvvisoDisplayed() {
         try {
             getWebDriverWait(5).withMessage("Il sezione scarica avviso non è visibile").until(ExpectedConditions.visibilityOf(scaricaAvviso)).isDisplayed();
             return true;
-        }catch (RuntimeException e){
+        } catch (RuntimeException e) {
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java]: as before

File: src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java [hunk 6]
```diff
@@ -829,6 +829,39 @@ public void verificaInvioNotificaDiCortesia() {
         }
     }
 
+    public void clickBottoneAnnullaNotifica() {
+        try {
+            WebElement bottoneAnnullaNotifica = driver.findElement(By.xpath("//button[@data-testid='cancelNotificationBtn']"));
+            getWebDriverWait(10).until(ExpectedConditions.and(ExpectedConditions.visibilityOf(bottoneAnnullaNotifica), ExpectedConditions.elementToBeClickable(bottoneAnnullaNotifica)));
+            scrollToElementAndClick(bottoneAnnullaNotifica);
+        } catch (TimeoutException e) {
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: as before

File: src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java [hunk 7]
```diff
@@ -829,6 +829,39 @@ public void verificaInvioNotificaDiCortesia() {
...
+            logger.error("Bottone annulla notifica non visibile e cliccabile");
+            Assert.fail("Bottone annulla notifica non visibile e cliccabile");
+        }
+    }
+
+    public void clickAnnullaNotificaModale() {
+        try {
+            By bottoneAnnullaNotificaModale = By.xpath("//button[@data-testid='modalCloseAndProceedBtnId']");
+            getWebDriverWait(10).withMessage("Bottone annulla notifica nella modale non visibile").until(ExpectedConditions.visibilityOfElementLocated(bottoneAnnullaNotificaModale));
+            getWebDriverWait(10).withMessage("Bottone annulla notifica nella modale non cliccabile ").until(ExpectedConditions.elementToBeClickable(bottoneAnnullaNotificaModale));
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: as before

File: src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java [hunk 8]
```diff
@@ -829,6 +829,39 @@ public void verificaInvioNotificaDiCortesia() {
...
+        } catch (TimeoutException e) {
+            logger.error("Bottone annulla notifica della modale non visibile e cliccabile");
+            Assert.fail("Bottone annulla notifica della modale non visibile e cliccabile");
+        }
+    }
+
+    public void checkPopUpConfermaAnnullamentoNotifica() {
+        try {
+            By popUpConfermaAnnullamento = By.xpath("//div[@role='alert']/div[text()='La richiesta di annullamento è stata accettata.']");
+            getWebDriverWait(10).withMessage("Pop up NON visualizzato").until(ExpectedConditions.visibilityOfElementLocated(popUpConfermaAnnullamento));
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: as before

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java (hunk 4 above)
Comment (leleOFA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java]: use assert for return boolean value

File: src/test/resources/feature/2-mittente/1_invioNotifiche/annullamentoNotifica.feature [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      | nomeCognomeDestinatario | Gaio Giulio      |
+      | codiceFiscale           | CSRGGL44L13H501E |
+    And Nella section Destinitario si clicca su "Aggiungi un indirizzo fisico" e si inseriscono i dati
+      | indirizzo | Via Roma |
+      | civico    | 20       |
+      | localita  | Milano   |
+      | comune    | Milano   |
+      | provincia | MI       |
+      | cap       | 20147    |
+      | stato     | Italia   |
```
Comment (leleOFA) [inline src/test/resources/feature/2-mittente/1_invioNotifiche/annullamentoNotifica.feature]: if create notification work, use here

File: src/test/resources/feature/2-mittente/3_downloadFile/014_1_downloadFileDettaglioNotificaMittente.feature (hunk 3 above)
Comment (Gabriiii99) [reply src/test/resources/feature/2-mittente/3_downloadFile/014_1_downloadFileDettaglioNotificaMittente.feature:14]: In blocked for now

File: src/test/resources/feature/2-mittente/1_invioNotifiche/annullamentoNotifica.feature (hunk 9 above)
Comment (Gabriiii99) [reply src/test/resources/feature/2-mittente/1_invioNotifiche/annullamentoNotifica.feature]: This test is duplicated, i don't know why is in this file. So i deleted this test

**Comment by Gabriiii99** [comment]
Build success on AWS

**Review by leleOFA — APPROVED** [review]

**Pull request merged**